pallets/parachain-staking/src/mock.rs

```diff
@@ -0,0 +1,728 @@
+// Copyright 2020-2022 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Test utilities
+use crate as pallet_parachain_staking;
+use crate::{
+    pallet, AwardedPts, Config, InflationInfo, Points, Range, COLLATOR_LOCK_ID, DELEGATOR_LOCK_ID,
+};
+use frame_support::{
+    construct_runtime, parameter_types,
+    traits::{Everything, GenesisBuild, LockIdentifier, OnFinalize, OnInitialize},
+    weights::Weight,
+};
+use sp_core::H256;
+use sp_io;
+use sp_runtime::{
+    testing::Header,
+    traits::{BlakeTwo256, IdentityLookup},
+    Perbill, Percent,
+};
+
+pub type AccountId = u64;
+pub type Balance = u128;
+pub type BlockNumber = u64;
+
+type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+type Block = frame_system::mocking::MockBlock<Test>;
+
+// Configure a mock runtime to test the pallet.
+// WHITELIST: Remove Session and CollatorSelection after end of whitelist-period
+construct_runtime!(
+    pub enum Test where
+        Block = Block,
+        NodeBlock = Block,
+        UncheckedExtrinsic = UncheckedExtrinsic,
+    {
+        System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+        Balances: pallet_balances::{Pallet, Call, Storage, Config<T>, Event<T>},
+        ParachainStaking: pallet_parachain_staking::{Pallet, Call, Storage, Config<T>, Event<T>},
+        BlockAuthor: block_author::{Pallet, Storage},
+        Session: pallet_session::{Pallet, Call, Storage, Event, Config<T>},
+        CollatorSelection: manta_collator_selection::{Pallet, Call, Storage, Config<T>, Event<T>},
+    }
+);
+
+parameter_types! {
+    pub const BlockHashCount: u64 = 250;
+    pub const MaximumBlockWeight: Weight = 1024;
+    pub const MaximumBlockLength: u32 = 2 * 1024;
+    pub const AvailableBlockRatio: Perbill = Perbill::one();
+    pub const SS58Prefix: u8 = 42;
```

Comment:
this isn't our prefix but probably doesn't matter for the mock

Reply:
postponed to https://github.com/Manta-Network/Manta/issues/798